docs/fabric-native-components.md

```diff
@@ -0,0 +1,217 @@
+```
+
+Next you'll need to create a folders structure to hold our `RTNCenteredText` component:
+
+```bash
+mkdir -p RTNCenteredText/{android,ios,js}
+```
+
+This gives you the following layout:
+
+```
+Demo
+├── android
+└── ios
+
+RTNCenteredText
+├── android
+├── ios
+└── js
+```
+
+## 1. Define Specification for Codegen
+
+Your specification must be defined in either [TypeScript](https://www.typescriptlang.org/) or [Flow](https://flow.org/) (see [Codegen](the-new-architecture/what-is-codegen) documentation for more details). This is used by Codegen to generate the C++, Objective-C++ and Java to connect your platform code to the JavaScript runtime that React runs in.
+
+Use this specification for our Text Component:
+
+<Tabs groupId="language" queryString defaultValue={constants.defaultJavaScriptSpecLanguage} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+The specification file must be named `<MODULE_NAME>NativeComponent.{ts,tsx}` to work with Codegen:
+
+```typescript title="RTNCenteredText/js/RTNCenteredTextNativeComponent.ts"
+import type {ViewProps} from 'react-native';
+import type {HostComponent} from 'react-native/Libraries/Renderer/shims/ReactNativeTypes';
+import type {Int32} from 'react-native/libraries/Types/CodegenTypes';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
+
+export interface NativeProps extends ViewProps {
+  text?: string;
+  margin?: Int32;
+  // add other props here
+}
+
+export default codegenNativeComponent<NativeProps>(
+  'RTNCenteredText',
+) as HostComponent<NativeProps>;
+```
+
+</TabItem>
+<TabItem value="flow">
+The specification file must be named `<MODULE_NAME>NativeComponent.{js,jsx}` to work with Codegen:
+
+```flow title="RTNCenteredText/js/RTNCenteredTextNativeComponent.js":
+// @flow strict-local
+
+import type {ViewProps} from 'react-native/Libraries/Components/View/ViewPropTypes';
+import type {HostComponent} from 'react-native';
+import type {Int32} from 'react-native/libraries/Types/CodegenTypes';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
+
+type NativeProps = $ReadOnly<{|
+  ...ViewProps,
+  text: ?string,
+  margin?: Int32,
+  // add other props here
+|}>;
+
+export default (codegenNativeComponent<NativeProps>(
+  'RTNCenteredText',
+): HostComponent<NativeProps>);
+```
+
+</TabItem>
+</Tabs>
+
+Add an `index.js` to wrap your
+
+As with Turbo Native Modules, you're able to have multiple specification files in the `js/` directory. For more information about the types you can use, and the platform types these map to see the [appendix](appendix.md#codegen-typings).
+
+## 2. Configure the Component for Codegen and Auto-Linking
+
+Typically Fabric Nactive Components are distributed using npm, it's also how our Codegen is configured. Add a [package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json) to `RTNCenteredText/package.json`:
```

Comment:
For consistency, we should use the same approach we used for TM. We should create them inside the app and then move them outside in a library.